inferentia/scripts/gen_triton_model.py

```diff
@@ -171,10 +200,96 @@ def initialize(self, args):
             instance_name_parts = args['model_instance_name'].split("_")
             if not instance_name_parts[-1].isnumeric():
                 raise pb_utils.TritonModelException(
-                    "internal error: the model instance name should end with \'_<instance_idx>\', got {}"
+                    "internal error: the model instance name should end with '_<instance_idx>', got {}"
                     .format(args['model_instance_name']))
             instance_idx = int(instance_name_parts[-1])
 
+        params = model_config['parameters']
+        compiled_model = params['COMPILED_MODEL']['string_value']
+        nc_start_idx = int(params['NEURON_CORE_START_INDEX']['string_value'])
+        nc_end_idx = int(params['NEURON_CORE_END_INDEX']['string_value'])
+        if nc_end_idx < nc_start_idx:
+            raise pb_utils.TritonModelException(
+                "the neuron core end index should be greater than or equal to the start index"
+            )
+
+        threads_per_core = int(params['NUM_THREADS_PER_CORE']['string_value'])
+        if threads_per_core < 1:
+            raise pb_utils.TritonModelException(
+                "the number of threads per core should be greater than or equal to 1"
+            )
+        num_threads = (nc_end_idx - nc_start_idx + 1) * threads_per_core
+
+        total_core_count = nc_end_idx - nc_start_idx + 1
+        if (instance_count > total_core_count):
+            raise pb_utils.TritonModelException(
+                "can not distribute {} triton model instances to {} neuron cores"
+                .format(instance_count, total_core_count))
+        cores_per_instance = total_core_count // instance_count
+'''
+    return init_impl
+
+def get_tensorflow_initialize_impl():
+    init_impl = get_common_initialize_impl()
+    init_impl += '''
+        self.input_list = []
+        for config_input in model_config['input']:
+            self.input_list.append(
+                (config_input['name'], config_input['data_type'],
+                 config_input['dims']))
+
+        self.output_list = []
+        for config_output in model_config['output']:
+            self.output_list.append(
+                (config_output['name'], config_output['data_type'],
+                 config_output['dims']))
+
+        # TODO: Validate input/output from the model
+
+        # TODO: NEURONCORE_GROUP_SIZES is deprecated by AWS Neuron
+        group_sizes = [str(1)] * cores_per_instance
+        os.environ["NEURONCORE_GROUP_SIZES"] = ','.join(group_sizes)
+
+        self.pred_list = [
+            tf.contrib.predictor.from_saved_model(compiled_model)
+            for _ in range(cores_per_instance)
+        ] * threads_per_core
+
+'''
+    return init_impl
+
+
```

Comment:
Not sure about formatting rules.. maybe this is one too many empty lines?

Reply:
Ran the formatter on the script.